sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/PartitionScopedRegionLevelProgress.java

```diff
@@ -0,0 +1,266 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.implementation;
+
+import com.azure.cosmos.CosmosException;
+import com.azure.cosmos.implementation.apachecommons.collections.map.UnmodifiableMap;
+import com.azure.cosmos.implementation.apachecommons.lang.StringUtils;
+import com.azure.cosmos.implementation.routing.RegionNameToRegionIdMap;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class PartitionScopedRegionLevelProgress {
+
+    private static final Logger logger = LoggerFactory.getLogger(PartitionScopedRegionLevelProgress.class);
+
+    private final ConcurrentHashMap<String, ConcurrentHashMap<String, RegionLevelProgress>> partitionKeyRangeIdToRegionLevelProgress;
+
+    private final ConcurrentHashMap<String, String> normalizedRegionLookupMap;
+
+    public final static String GlobalProgressKey = "global";
```

Comment:
nit: code convention not followed for static constant name.

Reply:
Fixed in next iteration.